students/PhongT/lesson06/mailroom.py

```diff
@@ -0,0 +1,176 @@
+        {"name": "Ken Lenning", "donations": [1325, 1232.24, 16325.5]},
+        {"name": "Thomas Timmy","donations": [521.3, 869.14]},
+        {"name": "Jane Chung", "donations": [3259.3, 1282.74, 1525.5]},
+        {"name": "Lucy Nguyen", "donations": [5253.82]},
+        {"name": "Ben Cormack", "donations": [56230.25, 89532.0, 12025.8]}]
+
+
+donors_data = get_donor_data()
+
+
+def print_thank_you():
+    """ Add or Update a donor record, then print a Thank You letter """
+    while True:
+        print()
+        response = input(''' sub_menu --- Donation Entries ---
+        Type your Full Name,
+        or 'list' to display list of the donor names,
+        or 'quit' to quit and return to main program prompt.
+        Enter your choice here: > ''')
+
+        if response.lower().strip() in ["q", "quit"]:
+            break
+        elif response.lower().strip() in ["l", "list"]:
+            print_donor_list()
+            continue
+        else:
+
+            amount = input("Enter Donation amount: ")
+            try:
+                donate_amount = float(amount)
+            except ValueError as e:
+                print("{} - The donation amount must be numeric!".format(e))
+            else:  # no exception occurs - nothing went wrong: add or update a record
+
+                record_id = find_donor_id(response)
+                donor_to_print = add_donor_to_db(response, donate_amount, record_id)
+
+                # now print a thank you letter
+                print(get_thankyou_message(donor_to_print))
+
+
+def find_donor_id(donor_name):
+    """
+    Find whether or not the input donor is in the db.
+    :param donor_name: the donor to search for in donor db
+    :return: donor key id if there is existing one in db. Otherwise return None
+    """
+    for index, existing_donor in enumerate(donors_data):
+        if existing_donor["name"].lower().strip() == donor_name.lower().strip():
+            return index
+    return None
+
+
+def add_donor_to_db(donor_name, amount, record_id=None):
+    """
+    Add a donor to the db.
+    :param donor_name: a donor to add
+    :param amount: donation amount
+    :param record_id: record id of an existing entry in db. Equals None if is new record
+    :return: a new added donor entry in db
+    """
+    if record_id is None:
+        logger.info('Add a new record entry')
+        donors_data.append({"name": donor_name, "donations": [amount]})
+        return donors_data[-1]
+    else:
+        logger.info('Update an existing record entry')
+        donors_data[record_id]["donations"].append(amount)
+        return donors_data[record_id]
+
+
+def get_thankyou_message(donor):
+    """ generate a formatted thank you letter message"""
+    message = '''Dear {}, 
+            Thank you so much for your generosity with your most recent donation of ${}. 
+            It will be put to very good use.
+            Sincerely.'''
+    return message.format(donor["name"], donor["donations"][-1])
```

Comment:
while nice, you probably want to keep the habit of doing your `.format()` with the string definition rather than the resulting variable